packages/editor/src/lib/editor/Editor.ts

```diff
@@ -5733,6 +5724,21 @@ export class Editor extends EventEmitter<TLEventMap> {
 				}
 			)
 
+			orderedShapeIds.forEach((originalId, i) => {
+				const originalShape = this.getShape(originalId)
+				if (!originalShape) return
```

Comment:
This should be an error, right? If there's no source shape then we shouldn't be duplicating it. As it is the code would leave all of these shapes with "a1" as their index.

Reply:
I updated the code so we don't rely on the order of the elements any longer.